### PR TITLE
Fix "invalid escape sequences" error on Python 3.6

### PR DESCRIPTION
**Pull request by yan12125**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Since Python 3.6, invalid escape sequences are invalid. An error occurs only if ```-W error``` is used. For example:
```
$ python -W error -m youtube_dl
Traceback (most recent call last):
  File "/usr/lib/python3.6/runpy.py", line 183, in _run_module_as_main
    mod_name, mod_spec, code = _get_module_details(mod_name, _Error)
  File "/usr/lib/python3.6/runpy.py", line 142, in _get_module_details
    return _get_module_details(pkg_main_name, error)
  File "/usr/lib/python3.6/runpy.py", line 109, in _get_module_details
    __import__(pkg_name)
  File "/home/yen/Projects/youtube-dl/youtube_dl/__init__.py", line 15, in <module>
    from .options import (
  File "/home/yen/Projects/youtube-dl/youtube_dl/options.py", line 8, in <module>
    from .downloader.external import list_external_downloaders
  File "/home/yen/Projects/youtube-dl/youtube_dl/downloader/__init__.py", line 3, in <module>
    from .common import FileDownloader
  File "/home/yen/Projects/youtube-dl/youtube_dl/downloader/common.py", line 10, in <module>
    from ..utils import (
  File "/home/yen/Projects/youtube-dl/youtube_dl/utils.py", line 504
    path_part if path_part in ['.', '..'] else re.sub('(?:[/<>:"\\|\\\\?\\*]|[\s.]$)', '#', path_part)
                                                             ^
SyntaxError: invalid escape sequence \s
```
This pull request fixes them all. Only broken strings are fixed. Regular expressions without invalid escape sequences are left untouched.

### CODE REVIEW

**Comment by yan12125** [comment]
References:

1. http://bugs.python.org/issue27364: Deprecate invalid escape sequences in str/bytes
2. http://bugs.python.org/issue25677: Syntax error caret confused by indentation

File: youtube_dl/extractor/bilibili.py [hunk 1]
```diff
@@ -82,10 +82,10 @@ def _real_extract(self, url):
                 'formats': formats,
             })
 
-        title = self._html_search_regex('<h1[^>]+title="([^"]+)">', webpage, 'title')
+        title = self._html_search_regex(r'<h1[^>]+title="([^"]+)">', webpage, 'title')
         description = self._html_search_meta('description', webpage)
         timestamp = unified_timestamp(self._html_search_regex(
-            r'<time[^>]+datetime="([^"]+)"', webpage, 'upload time', fatal=False))
+            '<time[^>]+datetime="([^"]+)"', webpage, 'upload time', fatal=False))
```
Comment (dstftw) [inline youtube_dl/extractor/bilibili.py]: Missing r?

Comment (yan12125) [inline youtube_dl/extractor/bilibili.py]: Thanks for brining up this. It's indeed a mistake.

First I replace all ```(?s)_search_regex\((\s*)'``` with ```_search_regex(\1r'``` and revert occurrences without invalid escape sequences. This ```r``` is accidentally removed during the revert phase.

Now a question is: should youtube-dl use raw strings for all regular expressions, or only for those that trigger errors on 3.6?

Comment (dstftw) [inline youtube_dl/extractor/bilibili.py]: As a rule of thumb, raw strings should be always used for regular expressions. But for now I guess it's enough to only fix ones with invalid escape sequences. At least this is what this PR's title stands for.

**Comment by yan12125** [comment]
Thanks for the feedback. Bilibili change reverted.

**Comment by dstftw** [comment]
LGTM.

**Comment by yan12125** [comment]
Landed as ec85ded.

**Pull request closed without merging**